### ISSUE TITLE
SQLite exception setting up IMAP account

### ISSUE THREAD

**Issue by tuxmachine** [Foundry376/Mailspring#816]
```txt
11790 [2018-04-08 09:41:19.015] [main] [info] Identity created at 1507273048 - using ID Schema 0
11790 [2018-04-08 09:41:19.017] [main] [info] ------------- Starting Sync (REDACTED) ---------------
11790 [2018-04-08 09:41:19.052] [metadata] [info] Metadata delta stream starting...
11790 [2018-04-08 09:41:19.053] [background] [info] Marking all folders as `busy`
11790 [2018-04-08 09:41:19.053] [background] [info] Syncing folder list...
11790 [2018-04-08 09:41:20.103] [background] [error] Created Mailspring container folder: Mailspring.
11790 [2018-04-08 09:41:20.262] [background] [error] Created required Mailspring folder: Mailspring/Snoozed.
11790 [2018-04-08 09:41:20.265] [background] [critical] 
 ***
 *** A C++ exception occurred during program execution: 
 *** UNIQUE constraint failed: Folder.id
 ***

11790 [2018-04-08 09:41:20.334] [background] [critical]  *** Stack trace (line numbers are approximate):
 *** in mailsync     SQLite::Statement::exec()
 *** in mailsync     MailStore::save(MailModel*, bool)
 *** in mailsync     SyncWorker::syncFoldersAndLabels()
 *** in mailsync     runBackgroundSyncWorker()
 *** in mailsync     main::$_4::operator()() const
 *** in mailsync     void* __1::__thread_proxy(void*)
 *** 0x7fff70e4d6c1  _pthread_body
 *** 0x7fff70e4d56d  _pthread_body
 *** 0x7fff70e4cc5d  thread_start
 ***
```

##### What operating system are you using?
OSX High Sierra

##### What version of Mailspring are you using?
1.2.1

--

**Bug?**
##### Do you have any third-party plugins installed? If so, which ones?
No

##### Is the issue related to a specific email provider (Gmail, Exchange, etc.)?
AFAIK it's a regular IMAP provider.

##### Is the issue reproducible with a particular attachment, message, signature, etc?
I've tried several versions all the way back to 1.0.12 - Deleted the `Application Support/Mailspring` directory each time - same error emerges.

**Comment by dilbertaz** [comment]
I am experiencing the same error with version 1.2.2-96fb3a99 on Ubuntu Linux 18.04 with Davmail 4.8.5, no plugins installed

**Comment by dprn** [comment]
Same error here, on MacOSX.

**Comment by tuxmachine** [comment]
Fixed in 1.5.5 👍

**Comment by CodeMouse92** [comment]
We are in the process of migrating issues to Discourse, which can better facilitate discussion and discovery, and so GitHub Issues can focus on issues that are confirmed and slated for resolution in the near term. [Learn more about the changes here.](https://community.getmailspring.com/t/mailspring-community-transition/38)

As part of this, we've migrated this issue to Discourse:

https://community.getmailspring.com/t/syncfolderuidrange-errorconnection-on-one-of-the-folders/412/2

Please consider joining that community and continuing the discussion there!

We're closing and locking the issue here as part of this migration. Rest assured, this doesn't mean the issue is being discarded or ignored.

We hope to see you on Discourse soon!

-The Mailspring Team
